Fixed some design issues and added few features (see commits)

ExtReMLapin opened the pull request:
## Added features : 

* Setting allowing to set the time to zero after pressing the play key (Start playing a song, pause it and un-pause it, it will replay it from the start)

* Anti-annoyance setting preventing the software from "typing" to the game when pressing any modifier key or when in a menu (like when typing in the chat)

* Auto install path detection (on first launch only)


And fixes by @p1ho

ExtReMLapin committed:
Added setting to set time to zero on play

ExtReMLapin committed:
fixed tabindexes of settigns and mainframe

ExtReMLapin committed:
updated default config

ExtReMLapin committed:
added checks to stop fuckingup the game

ExtReMLapin committed:
Anti-Annoyance is now in settings

ExtReMLapin committed:
Added auto installation path detection

Also removed the useless sanford dependency, just link to the assembly found in the release

ExtReMLapin committed:
Fixed extra backslash and junk debug messagebox

ExtReMLapin committed:
oops

p1ho commented on LuteBot/IO/KB/ActionManager.cs:
```diff
@@ -139,14 +139,17 @@ private static void InputCommand(int noteId)
-                return;
+            if (ConfigManager.GetBooleanProperty(PropertyItem.DontPlayNoteWhenRequired) == true)
+            {
+                if (pci.flags == CURSOR_SHOWING)
```
I hope this feature gets merged soon!
Question, when user does not have required state (like typing), does it pause the bot from playing?
If not, it might be nice to pause the player so they can resume playing the song where they left off?

ExtReMLapin replied:
At the start i only wanted to release a version that would reset the song after resuming it, i'm too lazy to make something good enough, i know it would be much better to do what you said.

ExtReMLapin replied:
It i guess we could backtrack the call branch and put the if statement at a better place, a place where the music would be implicitly paused, but i'm lazy, will maybe do it later.


If you want to, make a pull request on my repo and it will be visible here

p1ho replied:
I'll see what I can do!

p1ho replied:
I've made progress but I need help...

I wonder if there's a way for the program to know if the console is open?
The problem is that the cursor also shows when the console is open (aka when the note is being entered). So even in your current version, if you add `Console.WriteLine(pci.flags == CURSOR_SHOWING);` right before the block, you'll see that there's a lot of false positives.

In your case, because it doesn't actually stop the player, it's less noticeable, but in my case, it actually pauses on every false positive. Is there a reliable way for me to detect whether the console is open? I took a look at `consoleOn` in the same file, does not look promising.

ExtReMLapin replied:
I don't remember having any issues playing song tho

ExtReMLapin replied:
Oh you meant with your changes

p1ho replied:
I meant both.

If the song is not very complex (like no chords), then it won't be an issue because the ingame console would have closed long before the next note hits. I'm talking about cases where a note hits before the previous one closed the console.

ExtReMLapin replied:
I used this method because it's not really intrusive, i could read the memory from the game to check if the chatbox is being used, but VAC would probably not be happy with it.

p1ho replied:
[the entertainer](https://github.com/MontyLeGueux/Lutebot-2/blob/master/dump/the_entertainer.mid) in the dump file should demonstrate the problem after the intro.

After adding `Console.WriteLine(pci.flags == CURSOR_SHOWING);` before the block, ideally it should be always false if users don't do anything. But when the song gets faster and more complex, you'll see a few trues in the output.

p1ho replied:
Ah, I forgot one thing to try. I'll try to set my own instance variable that turns on and off when the program sends the 'open console' command and 'enter' command, see how that goes.

p1ho replied:
@MontyLeGueux 
I got a solution working. The main idea is knowing that console could still be up when `CURSORINFO()` first checks, we give it a very brief timeout (9ms is the smallest value that it works on my machine without false positives), and check it again. If the cursor is still up after the timeout, we assume it's user initiated.

here's an ugly snippet:
```C#
CURSORINFO pci = new CURSORINFO();
pci.cbSize = Marshal.SizeOf(typeof(CURSORINFO));
GetCursorInfo(ref pci); // it stores the cursordata to the struct

/*
    * If the cursor is showing, it means the user is typing or doesn't have the required state to play lute.
    */
if (ConfigManager.GetBooleanProperty(PropertyItem.DontPlayNoteWhenRequired) == true)
{
    if (pci.flags == CURSOR_SHOWING)
    {
        Thread.Sleep(9);
        CURSORINFO pci1 = new CURSORINFO();
        pci1.cbSize = Marshal.SizeOf(typeof(CURSORINFO));
        GetCursorInfo(ref pci1); // it stores the cursordata to the struct
        if (pci1.flags == CURSOR_SHOWING)
        {
            Console.WriteLine("Cursor still showing");
            return;
        }
    }
```

p1ho replied:
I've opened a PR to your fork [here](https://github.com/ExtReMLapin/mordhau-Lutebot/pull/1)

ExtReMLapin replied:
Thanks 👍  Merged all of it to this PR

p1ho committed:
Made it so player is paused rather than mute when conditions meet in-game

p1ho committed:
Changed variable name and setting text

p1ho committed:
Added 2 checks mechanism for Cursor Showing to decrease false positives

p1ho committed:
Changed the config Item PauseWhenRequired:
 - Checked: Pause when user initiated events detected
 - Uncheck: Mute instead of Pause, will resume playing automatically

ExtReMLapin committed:
Merge pull request #1 from p1ho/feature-autopause

Feature autopause

ExtReMLapin committed:
Create README.md

ExtReMLapin commented:
I found out what was causing issues on your side, i'm on a 144hz monitor and that's what was preventing the issue you had to happen on my side.

ExtReMLapin committed:
Update README.md

ExtReMLapin committed:
fixed crash when pressing dot on numpad, hotfix, not a design fix
